### PR TITLE
[#51239] 13.1 Documentation: sharing work packages with project non-m…

### PR DESCRIPTION
**Pull request by MayaBerd**
…embers  https://community.openproject.org/work_packages/51239

[#51239] 13.1 Documentation: sharing work packages with project non-members

https://community.openproject.org/work_packages/51239

### CODE REVIEW

File: docs/user-guide/work-packages/add-work-packages-to-meetings/README.md [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-  priority: 961
+  priority: 965
 description: How to add work packages to meetings agenda.
 keywords: meetings, dynamic meetings, meeting agenda
 ---
 
 # Add work packages to meetings
 
-Since the 13.1 Release work packages can be added directly to the meetings. This can be done either from the detailed view of a work package or from the [detailed view of a meeting](../../meetings/dynamic-meetings#add-a-work-package-to-the-agenda).
+Since the 13.1 release work packages can be added directly to the meetings. This can be done either from the detailed view of a work package or from the [detailed view of a meeting](../../meetings/dynamic-meetings#add-a-work-package-to-the-agenda).
```
Comment (marcalcobe) [inline docs/user-guide/work-packages/add-work-packages-to-meetings/README.md]: @MayaBerd I think we wanted to get rid of saying the release (13.1) as this then feels irrelevant after a couple of releases.

File: docs/user-guide/work-packages/share-work-packages/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+sidebar_navigation:
+  title: Share work packages
+  priority: 963
+description: How to share work packages in OpenProject.
+keywords: work package, share, share work packages
+---
+
+# Share work packages
+
+Since the 13.1 it is possible to share work packages with users that are not members of a project or are not yet registered on your instance. In the latter case a user will need to register in order to view the work package.
```
Comment (marcalcobe) [inline docs/user-guide/work-packages/share-work-packages/README.md]: @MayaBerd same here with the release number (13.1). Do we really want to state it?

File: docs/user-guide/work-packages/share-work-packages/README.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+> **Note**: Sharing work packages with non-member is an Enterprise add-on and can only be used with [Enterprise cloud](../../../enterprise-guide/enterprise-cloud-guide/) or  [Enterprise on-premises](../../../enterprise-guide/enterprise-on-premises-guide/). An upgrade from the free community edition is easy and helps support OpenProject.
+
+To share a work package with a project non-member select the detailed view of a work package and click the **Share** button.
+
+![Share button in OpenProject work packages](openproject_user_guide_share_button_wp.png)
+
+A dialogue window will open, showing the list of all users, who this work package has already been shared with. If the work package has not yet been shared, the list will empty.
```
Comment (marcalcobe) [inline docs/user-guide/work-packages/share-work-packages/README.md]: I think there is a "be" missing in the "will be empty."

File: docs/user-guide/work-packages/share-work-packages/README.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+> **Note**: Sharing work packages with non-member is an Enterprise add-on and can only be used with [Enterprise cloud](../../../enterprise-guide/enterprise-cloud-guide/) or  [Enterprise on-premises](../../../enterprise-guide/enterprise-on-premises-guide/). An upgrade from the free community edition is easy and helps support OpenProject.
+
+To share a work package with a project non-member select the detailed view of a work package and click the **Share** button.
+
+![Share button in OpenProject work packages](openproject_user_guide_share_button_wp.png)
+
+A dialogue window will open, showing the list of all users, who this work package has already been shared with. If the work package has not yet been shared, the list will empty. 
+
+> **Note**: For sharing with non members to be available, a respective role needs to be assigned to the **Non member role**.
```
Comment (marcalcobe) [inline docs/user-guide/work-packages/share-work-packages/README.md]: I'm not sure what this phrase wants to say. Is this about giving a share role to the external user?

File: docs/user-guide/work-packages/share-work-packages/README.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+If the list contains multiple users you can filter it by Type or Role. 
+
+Following user types are available as filters:
+
+![Filter list of users by user type](openproject_user_guide_sharing_member_type_filter.png)
+
+- Project member - returns all users that are project members
+- Not project member - returns all users that are not project members
+- Project group - returns all users that are members of the group, which includes the project
```
Comment (marcalcobe) [inline docs/user-guide/work-packages/share-work-packages/README.md]: I would change the "which includes the project" to something like "members of a group which is part of the project."

File: docs/user-guide/work-packages/share-work-packages/README.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+- Not project group - returns all users that are not members of the group, which includes the project
```
Comment (marcalcobe) [inline docs/user-guide/work-packages/share-work-packages/README.md]: I would also use the same format suggested for Project group

File: docs/user-guide/work-packages/share-work-packages/README.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+Following user roles are available as filters:
+
+![Filter list of users by user role](openproject_user_guide_sharing_member_role_filter.png)
+
+You can search for a user via a user name or an email address. You can either select an existing user from the dropdown menu or enter an email address for an entirely new user, who will receive an invitation to create an account on your instance.
```
Comment (marcalcobe) [inline docs/user-guide/work-packages/share-work-packages/README.md]: I would add "search for a user or group".

File: docs/user-guide/work-packages/share-work-packages/README.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+![search for a new user to share a work package](openproject_user_guide_shared_search.png)
+
+A user with whom you shared the work package will be added to the members of the project with the role **Work Package Viewer** and will be visible in the list of the project members. IS THIS TRUE? THEN WHY ARE THERE NON-PROJECT MEMBER ROLES AVAILABLE AS FILTERS?
```
Comment (marcalcobe) [inline docs/user-guide/work-packages/share-work-packages/README.md]: I don't think so. @HDinger can we confirm this?

File: docs/user-guide/work-packages/share-work-packages/README.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+![Filter list of users by user type](openproject_user_guide_sharing_member_type_filter.png)
+
+- Project member - returns all users that are project members
+- Not project member - returns all users that are not project members
+- Project group - returns all users that are members of the group, which includes the project
+- Not project group - returns all users that are not members of the group, which includes the project
+
+Following user roles are available as filters:
+
+![Filter list of users by user role](openproject_user_guide_sharing_member_role_filter.png)
```
Comment (marcalcobe) [inline docs/user-guide/work-packages/share-work-packages/README.md:45]: I don't know if we want to specify the capacities of each role (what the user can do with each of them). Opinions @HDinger?

File: docs/user-guide/work-packages/share-work-packages/README.md [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+  priority: 963
+description: How to share work packages in OpenProject.
+keywords: work package, share, share work packages
+---
+
+# Share work packages
+
+Since the 13.1 it is possible to share work packages with users that are not members of a project or are not yet registered on your instance. In the latter case a user will need to register in order to view the work package.
+
+> **Note**: Sharing work packages with non-member is an Enterprise add-on and can only be used with [Enterprise cloud](../../../enterprise-guide/enterprise-cloud-guide/) or  [Enterprise on-premises](../../../enterprise-guide/enterprise-on-premises-guide/). An upgrade from the free community edition is easy and helps support OpenProject.
```
Comment (marcalcobe) [inline docs/user-guide/work-packages/share-work-packages/README.md:13]: Maybe we need to say somewhere also that you need to have the correct rights to share work packages.

Comment (marcalcobe) [reply docs/user-guide/work-packages/share-work-packages/README.md:13]: If the user doesn't have the correct rights they will not be able to add users in the share modal (only view)

File: docs/user-guide/work-packages/share-work-packages/README.md (hunk 7 above)
Comment (marcalcobe) [reply docs/user-guide/work-packages/share-work-packages/README.md]: Maybe is also worth saying that you can add more than user/group at the same time.

**Review by marcalcobe — CHANGES_REQUESTED** [review]
Hi @MayaBerd,
I've added some comments to my review.

File: docs/user-guide/work-packages/share-work-packages/README.md (hunk 8 above)
Comment (HDinger) [reply docs/user-guide/work-packages/share-work-packages/README.md]: The sentence is half true. A shared user (who was not member of the project before) will be visible on the members page of the project with the role `Work Package Viewer` (or `[...] Editor` or `[...] Commenter`). The user can be seen as member of a specific WorkPackage within the project, that is why it is listed there. 
However this user is **not** automatically a member of the whole project. A member usually has much more permissions than a person who should just see a specific Work Package. Thus, the filter makes sense. 

Imho, we should rethink about listing the shared users on the members page, at it might create more confusion than help. In the next version, we could for exampe split the page up (e.g. via tabs) into "normal" members and "work package" members.

File: docs/user-guide/work-packages/share-work-packages/README.md (hunk 9 above)
Comment (HDinger) [reply docs/user-guide/work-packages/share-work-packages/README.md:45]: It probably makes sense to explain the capacitites briefly. In English, there are pretty self-explanatory, but in other languages the translations might not as intuitive. (e.g. the german translation for "view" is currently "Ansicht" which even I did not understand directly). 

It might be worth to mention that filters show you only the explicitly granted permissions, not the inherited ones. 
So imagine a user who is shared with permission "view" and the user is part of a group which is shared with as well with permission "comment" . When I filter for "comment", I will only see the group, although the user inherited that permission as well. This is indicated via the status text below the group but it might be worth mentioning nevertheless.

File: docs/user-guide/work-packages/add-work-packages-to-meetings/README.md (hunk 1 above)
Comment (MayaBerd) [reply docs/user-guide/work-packages/add-work-packages-to-meetings/README.md]: good point, i will remove the mention of the release

**Comment by MayaBerd** [comment]
@marcalcobe thank you for the review. and @HDinger thank you for your helpful comments! 
I've adjusted the text accordingly.
